From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 14 Mar 2018 00:40:53 +0000
Subject: x86/cpu: Avoid ABI change in 4.15.5
Forwarded: not-needed

The type of cpuinfo_x86::x86_cache_size was changed from int to
unsigned int, and the "unknown" value was changed from -1 to 0.  Only
one in-tree user of the field seems to have checked for the "unknown"
value, so this is probably not an ABI change in practice.  Just hide
the type change from genksyms.

---
--- a/arch/x86/include/asm/processor.h
+++ b/arch/x86/include/asm/processor.h
@@ -109,7 +109,11 @@ struct cpuinfo_x86 {
 	char			x86_vendor_id[16];
 	char			x86_model_id[64];
 	/* in KB - valid for CPUS which support this call: */
+#ifndef __GENKSYMS__
 	unsigned int		x86_cache_size;
+#else
+	int			x86_cache_size;
+#endif
 	int			x86_cache_alignment;	/* In bytes */
 	/* Cache QoS architectural values: */
 	int			x86_cache_max_rmid;	/* max index */
